File: single-language-analysis/dataset-files/java/TestSmell021.java
// Original URL: https://github.com/vigusmao/Comp2_2020_PLE/blob/f38a995ff51ade221cdca4c6b72b98b88f13d9ef/Siguinha/src/testes/AlunoTest.java#L48-L84

public class TestSmell021 {
    @Test
    public void getConclusaoDiscipinaTest() {
        ArrayList<ItemDeHistorico> historico = aluno.getHistorico();
        assertNotNull(historico);
        for (int i = 0; i < historico.size(); i++) {
            assertNull(historico.get(i));
        }
        assertEquals(0, aluno.getQuantDisciplinasCursadas());

        // o aluno vai cursar a primeira disciplina
        aluno.registrarConclusaoDisciplina(disciplina1, 6.5f, 2019, 2);

        historico = aluno.getHistorico();
        ItemDeHistoricoDisciplinaCursada primeiroItem = (ItemDeHistoricoDisciplinaCursada) historico.get(0);
        assertEquals("MAB001", primeiroItem.getDisciplina().getCodigo());
        assertEquals(2019, primeiroItem.getAno());
        assertEquals(2, primeiroItem.getSemestre());
        verificarAtualizacaoCreditos(1, 6.5f, 4);

        aluno.registrarConclusaoDisciplina(disciplina2, 8, 2020, 1);
        // verificar todas as disciplinas do histórico até aqui
        historico = aluno.getHistorico();
        assertEquals("MAB001", primeiroItem.getDisciplina().getCodigo());
        ItemDeHistoricoDisciplinaCursada segundoItem = (ItemDeHistoricoDisciplinaCursada) historico.get(1);
        assertEquals("MAB002", segundoItem.getDisciplina().getCodigo());
        verificarAtualizacaoCreditos(2, 7.4f, 10);

        // o aluno vai cursar a terceira disciplina
        aluno.registrarConclusaoDisciplina(disciplina3, 10, 2020, 1);

        historico = aluno.getHistorico();
        assertEquals("MAB001", primeiroItem.getDisciplina().getCodigo());
        assertEquals("MAB002", segundoItem.getDisciplina().getCodigo());
        ItemDeHistoricoDisciplinaCursada terceiroItem = (ItemDeHistoricoDisciplinaCursada) historico.get(2);
        assertEquals("MAJ003", terceiroItem.getDisciplina().getCodigo());
        verificarAtualizacaoCreditos(3, 8.375f, 16);
    }
}

File: single-language-analysis/dataset-files/python/test_smell_021.py
import pytest

class TestAluno:
    def test_smell_021(self):
        historico = aluno.get_historico()
        assert historico is not None
        for i in range(len(historico)):
            assert historico[i] is None
        assert aluno.get_quant_disciplinas_cursadas() == 0

        aluno.registrar_conclusao_disciplina(disciplina1, 6.5, 2019, 2)

        historico = aluno.get_historico()
        primeiro_item = historico[0]
        assert primeiro_item.get_disciplina().get_codigo() == "MAB001"
        assert primeiro_item.get_ano() == 2019
        assert primeiro_item.get_semestre() == 2
        self.verificar_atualizacao_creditos(1, 6.5, 4)

        aluno.registrar_conclusao_disciplina(disciplina2, 8, 2020, 1)
        historico = aluno.get_historico()
        assert primeiro_item.get_disciplina().get_codigo() == "MAB001"
        segundo_item = historico[1]
        assert segundo_item.get_disciplina().get_codigo() == "MAB002"
        self.verificar_atualizacao_creditos(2, 7.4, 10)

        aluno.registrar_conclusao_disciplina(disciplina3, 10, 2020, 1)

        historico = aluno.get_historico()
        assert primeiro_item.get_disciplina().get_codigo() == "MAB001"
        assert segundo_item.get_disciplina().get_codigo() == "MAB002"
        terceiro_item = historico[2]
        assert terceiro_item.get_disciplina().get_codigo() == "MAJ003"
        self.verificar_atualizacao_creditos(3, 8.375, 16)

    def verificar_atualizacao_creditos(self, arg1, arg2, arg3):
        pass